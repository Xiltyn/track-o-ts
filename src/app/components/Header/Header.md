Header example:

```js
<Header
    message="Hello Styleguidist!"
    logotype={ <img src={ require('../../../assets/images/DND_LOGO.png') } alt="Brand logotype"/> }/>
```
