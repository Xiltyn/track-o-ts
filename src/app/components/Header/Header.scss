@import "../../style/core/variables";
@import "../../style/utils/functions";
@import "../../style/utils/mixins";

.app-header {
    position: fixed;
    top: 0;
    left: 0;
    height: 80px;
    width: 100%;
    padding: 12px;

    color: $GREY_LIGHT;
    display: flex;
    user-select: none;
    align-items: center;
    justify-content: space-between;

    background: $GREY_DARK;
    box-shadow: 0 0 24px -4px #000;
    box-sizing: border-box;

    z-index: 50;

    .logo {
        color: $GREY_TEXT;
        pointer-events: none;
    }

    div {
        position: relative;

        width: 56px;
        height: 34px;
        cursor: pointer;

        svg {
            width: 100%;
            height: 100%;

            path {
                fill: $GREY;
                transition: all .3s ease-in-out;
            }
        }

        &:hover {
            svg path {
                fill: $GREY_TEXT;
            }
        }
    }
}
