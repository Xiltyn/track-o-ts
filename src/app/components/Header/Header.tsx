import * as React from 'react';

import './Header.scss';

/**
 * Header Properties.
 */
export namespace Header {
    export interface Props {
        /** Brand logotype image */
        logotype:React.ReactElement<SVGElement|HTMLElement>;
        /** Example header message prop */
        message:string;
    }

    export interface State {
        /** Current examples message set on Header */
        currentMessage:string,
    }
}

/**
 * Application Header.
 */
export class Header extends React.Component<Header.Props, Header.State> {
    constructor(props:Header.Props) {
        super(props);

        this.state = {
            currentMessage: this.props.message,
        }
    }

    switchMessage = (text:string) => {
        if(text) {
            this.setState({ currentMessage: text });
        }
    };

    render() {
        const {
            logotype,
        } = this.props;

        const {
            currentMessage,
        } = this.state;

        return (
            <header
                className="app-header"
                onClick={ () => this.switchMessage('Don\' touch this!' ) }>
                <div className="logotype-container">
                    { logotype }
                </div>
                <h3 className="header-message">
                    { currentMessage }
                </h3>
            </header>
        );
    }
}
