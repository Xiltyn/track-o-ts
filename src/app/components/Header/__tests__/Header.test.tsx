import * as React from 'react';
import { shallow } from 'enzyme';
import { Header } from './../Header';

const wrapper = shallow(<Header
    logotype={ require('assets/images/DND_LOGO.png') }
    message='Welcome to Dev&Deliver boilerplate for React apps with TypeScript!'/>
);

describe('<|:===== <Header /> =====:|>', () => {
    it('renders component', () => {
        expect(wrapper).toMatchSnapshot();
    });
});

