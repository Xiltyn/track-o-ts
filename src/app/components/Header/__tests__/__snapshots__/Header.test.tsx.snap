// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<|:===== <Header /> =====:|> renders component 1`] = `
<header
  className="app-header"
  onClick={[Function]}
>
  <div
    className="logotype-container"
  >
    test-file-stub
  </div>
  <h3
    className="header-message"
  >
    Welcome to Dev&Deliver boilerplate for React apps with TypeScript!
  </h3>
</header>
`;
